whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/FullTextSearchService.java

```diff
@@ -65,6 +70,23 @@ public Response search(
         }
     }
 
+    private static String escapeColon(final String query){
+        if (StringUtils.isEmpty(query) || !query.contains(":") || query.contains("\\:")){
+            return query;
+        }
+
+        final StringBuilder sb = new StringBuilder();
+
+        final String[] split = query.split(":");
+        Arrays.stream(split).limit(split.length - 1).forEach(splitColon -> sb.append(splitColon).append(shouldEscapeColon(splitColon) ? "\\:" : ":"));
+        return sb.append(split[split.length - 1]).toString();
+    }
+
+    private static boolean shouldEscapeColon(final String splitColon) {
```

Comment:
Since we are using the same api for public queries too, can we also  test the integration with db-web-ui too ? It should not impact that.